src/event.c

```diff
@@ -267,10 +278,23 @@ subject_attr_t *get_subj_attr(event_t *e, subject_type_t t)
 		return sn;
 	}
 
-	free(subj.str);
+	// free .str only when it was really used
+	// otherwise invalid free is possible
+	if (t >= COMM) free(subj.str);
 	return NULL;
 }
 
+/*
+ * If subject is trusted function return true, false otherwise.
+ */
+int is_subj_trusted(event_t *e)
+{
+  subject_attr_t *trusted = get_subj_attr(e, SUBJ_TRUST);
+
+  if (!trusted) return 0;
+  return trusted->val;
+}
```

Comment:
Why is this not just:

```
bool is_subj_trusted(event_t *e)
{
        subject_attr_t *trusted = get_subj_attr(e, SUBJ_TRUST);

        return trusted ? true : false;
}
```
Or, if you're relying on the `int` value returned in `trusted->val` then just return that instead and change the function signature...

Reply:
To be honest, I think it'd make more sense to move this helper into `src/rules.c` and have it:

```
static bool is_subj_trusted(...)
{
...
}
```

You're already exposing `get_subj_attr()` anyway, and based on this commit it doesn't look like it's being used anywhere outside of `src/rules.c`?

Reply:
> Why is this not just:
> 
> ```
> bool is_subj_trusted(event_t *e)
> {
>         subject_attr_t *trusted = get_subj_attr(e, SUBJ_TRUST);
> 
>         return trusted ? true : false;
> }
> ```
> 
> Or, if you're relying on the `int` value returned in `trusted->val` then just return that instead and change the function signature...

Yes, return type can be bool but ```trusted``` itself can be NULL so we need to check this before accessing the```val``` attribute.

Reply:
Sure. You should also consider putting it in `src/rules.c`. Makes more sense.

Reply:
Done.